src/java_bytecode/java_object_factory.cpp

```diff
@@ -513,31 +536,56 @@ Function: object_factory
 
 exprt object_factory(
   const typet &type,
+  const irep_idt base_name,
   code_blockt &init_code,
   bool allow_null,
   symbol_tablet &symbol_table,
   size_t max_nondet_array_length,
   const source_locationt &loc)
 {
-  symbolt &aux_symbol=new_tmp_symbol(
-    symbol_table,
-    loc,
-    type);
-  aux_symbol.is_static_lifetime=true;
+  irep_idt identifier=id2string(goto_functionst::entry_point())+
+    "::"+id2string(base_name);
+
+  auxiliary_symbolt main_symbol;
+  main_symbol.mode=ID_C;
+  main_symbol.is_static_lifetime=false;
+  main_symbol.name=identifier;
+  main_symbol.base_name=base_name;
+  main_symbol.type=type;
+  main_symbol.location=loc;
+
+  exprt object=main_symbol.symbol_expr();
 
-  exprt object=aux_symbol.symbol_expr();
+  symbolt *main_symbol_ptr;
+  bool moving_symbol_failed=symbol_table.move(main_symbol, main_symbol_ptr);
+  assert(!moving_symbol_failed);
+
+  std::vector<symbolt const *> symbols_created;
+  symbols_created.push_back(main_symbol_ptr);
 
   java_object_factoryt state(
-    init_code,
+    symbols_created,
+    loc,
     !allow_null,
     max_nondet_array_length,
     symbol_table);
+  code_blockt assignments;
   state.gen_nondet_init(
+    assignments,
     object,
     false,
     "",
-    loc,
     false);
 
+  // Add the following code to init_code for each symbol that's been created:
+  //   <type> <identifier>;
+  for(symbolt const *symbol_ptr : symbols_created)
```

Comment:
I think `const symbolt *symbol_ptr` is more consistent, and I think it can even be a `const symbolt * const`